apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+            ConversationView.cancelEdit();
+            WaitingScreen.hide();
+          });
+        }
+      );
+    }
+
+    return Utils.confirm(
+      {
+        id: 'deleteMessages-confirmation-message',
+        args: { n: this.selectionHandler.selectedCount }
+      },
+      null,
+      {
+        text: 'delete',
+        className: 'danger'
+      }
+    ).then(performDeletion.bind(this));
+  },
+
+  cancelEdit: function conv_cancelEdit() {
+    if (this.inEditMode) {
+      this.inEditMode = false;
+      this.mainWrapper.classList.remove('edit');
+    }
+  },
+
+  updateSelectionStatus: function conv_updateSelectionStatus() {
+    var selected = this.selectionHandler.selectedCount;
+
+    // Manage buttons enabled\disabled state
+    if (this.selectionHandler.allSelected()) {
+      this.checkUncheckAllButton.setAttribute('data-l10n-id', 'deselect-all');
+    } else {
+      this.checkUncheckAllButton.setAttribute('data-l10n-id', 'select-all');
+    }
+
+    if (selected > 0) {
+      this.deleteButton.disabled = false;
+      navigator.mozL10n.setAttributes(this.editMode, 'selected-messages',
+        {n: selected});
+    } else {
+      this.deleteButton.disabled = true;
+      navigator.mozL10n.setAttributes(this.editMode, 'deleteMessages-title');
+    }
+  },
+
+  handleMessageClick: function conv_handleMessageClick(evt) {
+    var currentNode = evt.target;
+    var elems = {};
+
+    // Walk up the DOM, inspecting all the elements
+    while (currentNode && currentNode.classList) {
+      if (currentNode.classList.contains('bubble')) {
+        elems.bubble = currentNode;
+      } else if (currentNode.classList.contains('message')) {
+        elems.message = currentNode;
+      } else if (currentNode.classList.contains('message-status')) {
+        elems.messageStatus = currentNode;
+      }
+      currentNode = currentNode.parentNode;
+    }
+
+    // Click event handlers that occur outside of a message element should be
+    // defined elsewhere.
+    if (!(elems.message && elems.bubble)) {
+      return;
+    }
+
+    // handle not-downloaded messages
+    if (elems.message.classList.contains('not-downloaded')) {
+
+      // do nothing for pending downloads, or expired downloads
+      if (elems.message.classList.contains('expired') ||
+        elems.message.classList.contains('pending')) {
+        return;
+      }
+      this.retrieveMMS(elems.message);
+      return;
+    }
+
+    // Do nothing for invalid empty content error because it's not possible to
+    // retrieve message again in this edge case.
+    if (elems.message.classList.contains('invalid-empty-content')) {
+      return;
+    }
+
+    // Click events originating from a "message-status" aside of an error
+    // message should trigger a prompt for retransmission.
+    if (elems.message.classList.contains('error') && elems.messageStatus) {
+      Utils.confirm('resend-confirmation').then(() => {
+        this.re
```

Comment:
We'll likely have two separate and bit simplified ```sendMessage``` methods for every view, haven't tried to split it yet.